src/wallet/wallet.cpp

```diff
@@ -1139,6 +1163,82 @@ void CWallet::SyncTransaction(const CTransaction& tx, const CBlockIndex *pindex,
     }
 }
 
+void CWallet::UpdatedBlockHeaderTip(bool fInitialDownload, const CBlockIndex *pindexNew)
+{
+    int bestSPVBlockHeight = 0;
+    {
+        LOCK2(cs_main, cs_wallet);
+        bestSPVBlockHeight = nBestSpvHeight;
+    }
+
+    // check if the new tip extends the wallets current known best headers tip
+    const CBlockIndex *pCurrent = pindexNew;
+    bool fReorg = true;
+    while (pCurrent)
+    {
+        if (pCurrent->GetBlockHash() == bestSpvBlockHash || bestSpvBlockHash.IsNull())
+        {
+            // new header tip extends the current headers chain, standard operation
+            fReorg = false;
+            break;
+        }
+        pCurrent = pCurrent->pprev;
+    }
+    if (fReorg)
+    {
+        // we are on a different chain, we don't know at which dept the fork happend
+        // do a SPV rescan back to the oldest wtx time that was received from a block that now has been forked away
+        CWalletDB walletdb(strWalletFile, "r+", false);
+        int64_t rescanBackTo = std::numeric_limits<int64_t>::max();
+        // the current wallets best header tip is not in the active headers chain
+        for (map<uint256, CWalletTx>::iterator it = mapWallet.begin(); it != mapWallet.end(); ++it)
+        {
+            CWalletTx* wtx = &(*it).second;
+            if (wtx->fSPV)
```

Comment:
Why is it significant that the txn is SPV? Isn't any transaction that's been reorged away be no longer confirmed? Might be good to have a comment on this here.